test/Interop/Cxx/namespace/Inputs/templates-with-forward-decl.h

```diff
@@ -0,0 +1,37 @@
+#ifndef TEST_INTEROP_CXX_NAMESPACE_INPUTS_TEMPLATES_WITH_FORWARD_DECL_H
+#define TEST_INTEROP_CXX_NAMESPACE_INPUTS_TEMPLATES_WITH_FORWARD_DECL_H
+
+namespace NS1 {
+
+template <typename T>
```

Comment:
Come to think of it, I bet you could get this behavior without using a template. Can you please try that :)

Reply:
Hmm that's weird, but it doesn't seem reproducible without a template. I'll need a bit more time to figure out why.

Reply:
That is interesting. It would be good to understand why we need this to be a template. But in the mean time, I don't think this patch should be blocked on that.